src/cosl/coordinated_workers/worker.py

```diff
@@ -255,19 +259,26 @@ def status(self) -> ServiceEndpointStatus:
         # we really don't want this code to raise errors, so we blanket catch all.
         try:
             services = self._container.get_services(*layer.services.keys())
-            running_status = {name: svc.is_running() for name, svc in services.items()}
-            if not all(running_status.values()):
-                if any(running_status.values()):
-                    starting_services = tuple(
+            if services:
+                # avoid vacuous quantification
+                running_status = {name: svc.is_running() for name, svc in services.items()}
```

Comment:
Given the extra logic below, should we replace this with a plain for that populates two separate dicts, `active`, `stopped`? Perhaps an ops feature request?

Reply:
I wouldn't mind it being in ops, but I'm not sure it's general enough to make the bar. will see if the split makes the logic more clear

Reply:
turned out flipping the polarity of the check made the logic simpler, we don't need to track the 'started' services at all.